### PR TITLE
test(autosharding): Functional Tests

### PR DESCRIPTION
**Pull request by AlejandroCabeza**
# Description
Implement autosharding tests.

# Changes
- [X] Autosharding tests
- [X] Helper Table==Array 
- [X] Remove legacy tests¡

### CODE REVIEW

**Review by NagyZoltanPeter — CHANGES_REQUESTED** [review]
LGTM except the binary file added. Thanks for it.

**Review by NagyZoltanPeter — APPROVED** [review]
It's great, thank you!

File: tests/waku_core/topics/test_sharding.nim [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      # When calling with pubsubTopic as none with invalid content
+      let topicMap = parseSharding(PubsubTopic.none(), contentTopicInvalid)
+
+      # Then the topicMap is valid
+      check:
+        topicMap.error() == "Cannot autoshard content topic: Generation > 0 are not supported yet"
+
+    xtest "catchable error on add to topicMap":
+      # TODO: Trigger a CatchableError or mock
+      discard
```
Comment (SionoiS) [inline tests/waku_core/topics/test_sharding.nim:153]: IMO either remove, add an issue or do it.

**Review by SionoiS — APPROVED** [review]
LGTM!

Thanks.

File: tests/waku_core/topics/test_sharding.nim (hunk 1 above)
Comment (AlejandroCabeza) [reply tests/waku_core/topics/test_sharding.nim:153]: I've implemented a simple mocking mechanism in PR depending on this one. If you guys don't mind, I'd love for this to get merged first so there's no need to reorder the commits, and I'll finish this test case as soon as it's available.

**Pull request merged**